cluster/mock.go

```diff
@@ -10,15 +12,16 @@ import (
 
 // Doubles as a cluster.Cluster and cluster.Manifests implementation
 type Mock struct {
+	Scoper             Scoper
 	AllServicesFunc    func(maybeNamespace string) ([]Controller, error)
 	SomeServicesFunc   func([]flux.ResourceID) ([]Controller, error)
 	PingFunc           func() error
 	ExportFunc         func() ([]byte, error)
 	SyncFunc           func(SyncDef) error
 	PublicSSHKeyFunc   func(regenerate bool) (ssh.PublicKey, error)
 	UpdateImageFunc    func(def []byte, id flux.ResourceID, container string, newImageID image.Ref) ([]byte, error)
-	LoadManifestsFunc  func(base string, paths []string) (map[string]resource.Resource, error)
-	ParseManifestsFunc func([]byte) (map[string]resource.Resource, error)
+	LoadManifestsFunc  func(base string, paths []string, scoper Scoper) (map[string]resource.Resource, error)
```

Comment:
I didn't know this was supposed to be a cluster-independent abstraction, sorry. However, note that it has quite a few leaks (assuming there are manifests, and the namespaces in the resource ids as you said).

However, we only need the scope for namespace filtering, so, if we move that out of `LoadManifests` maybe we can also move scoping out of parsing. Note, however, that then we would need a way to get the original namespace of a resource (for scoping), not the one currently on the resource ID which can be synthesized if empty.

Reply:
It totally has leaks. It will probably (at this point) never be extended to cover other types of cluster. 

However, it's still worth trying to contain changes, where we can. The namespace scoping is particular to Kubernetes, so far as we know (and I wouldn't expect to see some other Scoping implementation be combined with a kubernetes cluster).